Form/Type/EnumType.php

```diff
@@ -63,23 +61,26 @@ public function setDefaultOptions(OptionsResolverInterface $resolver)
         };
 
         $enumClass = function(Options $options) {
-            if ($options->has('data') && is_object($enum = $options->get('data'))) {
+            if ($options->offsetExists('data') && is_object($enum = $options->offsetGet('data'))) {
                 return get_class($enum);
             }
 
             return null;
         };
 
-        $resolver->setDefaults(array(
-            'enum_class' => $enumClass,
-            'choices'    => $choices,
-        ));
-
-        $resolver->setAllowedTypes(array(
-            'enum_class' => 'string'
-        ));
+        $resolver
+            ->setDefined(array('enum_class', 'choices'))
```

Comment:
With new api, seems to be mandatory, you can see Travis. My first commits didn't have it, I had to add this.